bookkeeper-server/src/main/java/org/apache/bookkeeper/client/RackawareEnsemblePlacementPolicyImpl.java

```diff
@@ -678,6 +728,38 @@ protected BookieNode selectRandomFromRack(String netPath, Set<Node> excludeBooki
         throw new BKNotEnoughBookiesException();
     }
 
+    /**
+     * Sort the node list by the network location.
+     * The higher the frequency of the network location, it will on the list head.
+     */
+    private List<Node> sortByCommonRack(List<Node> nodeLists) {
+        Map<String, Integer> map = new HashMap<>();
+        for (Node node : nodeLists) {
+            map.put(node.getNetworkLocation(), map.getOrDefault(node.getNetworkLocation(), 0) + 1);
+        }
+        List<Object[]> list = new ArrayList<>();
```

Comment:
Do you need to set the length of the list in advance?

Reply:
The size is not unpredictable